com.unity.perception/Documentation~/Tutorial/Phase2.md

```diff
@@ -0,0 +1,222 @@
+
+The UI for `My Light Randomizer` should now look like this:
+
+<p align="center">
+<img src="Images/light_rand_2.png" width="420"/>
+</p>
+
+
+* **Action**: Run the simulation for a few frames to observe the lighting color changing on each iteration.
+
+
+### <a name="step-2">Step 2: Bundle Data and Logic Inside Randomization Tags</a> 
+
+
+You may sometimes need to bundle certain randomization-related data or logic within an object that are inherent to the object itself. For instance, you may have multiple lights in the Scene but would like each of them to have their own unique range of intensities. It would be quite tedious to add a new Parameter to your light Randomizer for each of your lights. Furthermore, this would make your light Randomizer excessively tailored to one use-case, limiting the Randomizer's reusability. 
+
+There are also cases were you may need to include certain logic within your object in order to make the Randomizer code more reusable and easy to maintain. For instance, you may want to build an office chair Prefab to use in various simulations. This chair is likely to support a range of customizations for its various parts (back angle, seat angle, seat height, etc.). Instead of directly mapping a Rotation Parameter from a Randomizer to the rotation of the back angle object within the chair, it might be more convenient to have the chair expose the range of possible angles in the form of a simple float between 0 and 1. With this approach, the Randomizer would only need to sample a float Parameter and assign it to the chair. The chair would in turn have a script attached that knows how to map this single float to a certain plausible back angle. You could even map this float to a more complex state of the chair. Your Randomizer would still only need one float Parameter.
+
+Let's try this approach with our `Directional Light` object. We will create a duplicate of this light and then have the two lights use different ranges of intensity while both using the exact same float Parameter from `MyLightRandomizer.cs`.
+
+* **Action**: Right-click on `Directional Light` in the Scene _**Hierarchy**_, and select _**Duplicate**_. The new light will automatically be named `Directional Light (1)`.
+* **Action**: Change the Y rotation of `Directional Light (1)` to 60, as shown below:
+
+<p align="center">
+<img src="Images/light_rand_2.png" width="420"/>
+</p>
+
+* **Action**: Change the Y rotation of `Directional Light` to -60.
+
+This makes the two lights illuminate the scene from opposing sides, each having a 30 degree angle with the background and foreground planes.
+
+* **Action**: Open `MyLightRandomizerTag.cs` and modify it to match the code below:
+```
+using UnityEngine;
+using UnityEngine.Experimental.Perception.Randomization.Randomizers;
+
+[AddComponentMenu("Perception/RandomizerTags/MyLightRandomizerTag")]
+public class MyLightRandomizerTag : RandomizerTag
```

Comment:
We can add a [RequireComponent(typeof(Light))] attribute to this class to ensure that there is a light component attached to any GameObject you add this MyLightRandomizerTag to.